annotations/views/rest_views.py

```diff
@@ -44,16 +45,19 @@ def has_permission(self, request, view):
         if not super().has_permission(request, view):
             return False
             
-        # Allow GET requests for authenticated users
         if request.method in ['GET', 'HEAD', 'OPTIONS']:
             return True
 
-        # check if user is owner or collaborator For POST/PUT/DELETE 
+        if request.method == 'DELETE':
+            return True  # Let has_object_permission handle it
+
+        if request.method in ['PUT', 'PATCH']:
+            return True  # Let has_object_permission handle it
```

Comment:
why not add all of the above to line 48? with a comment that includes the comments here?